py4DSTEM/process/diskdetection/diskdetection.py

```diff
@@ -566,5 +595,146 @@ def threshold_Braggpeaks(pointlistarray, minRelativeIntensity, relativeToPeak, m
+            pointlist.sort(coordinate='intensity', order='descending')
+            
+            if pointlist.data.shape[0] == 0:
+                top_value = np.nan
+                
+            else:
+                top_value = pointlist.data[0][2]
+                HI_array[Rx, Ry] = top_value
+    
+    mean_intensity = np.nanmean(HI_array)
+    max_intensity = np.max(HI_array)
+    median_intensity = np.median(HI_array)
+    
+    for (Rx, Ry) in tqdmnd(pointlistarray.shape[0],pointlistarray.shape[1]):
+            
+            pointlist = pointlistarray.get_pointlist(Rx,Ry)
+            #pointlist.sort(coordinate='intensity', order='descending')
+
+            # Remove peaks below minRelativeIntensity threshold
+            
+            if minIntensity is not False:
+                if metric == 'average':
+                    deletemask = pointlist.data['intensity'] / mean_intensity < \
+                                                                               minIntensity
+                    pointlist.remove_points(deletemask)
+                
+                if metric == 'maximum':
+                    deletemask = pointlist.data['intensity'] / max_intensity < \
+                                                                               minIntensity
+                    pointlist.remove_points(deletemask)
+                    
+                if metric == 'median':
+                    deletemask = pointlist.data['intensity'] / median_intensity < \
+                                                                               minIntensity
+                    pointlist.remove_points(deletemask)
+                    
+                if metric == 'manual':
+                    deletemask = pointlist.data['intensity'] < minIntensity
+                    
+                    pointlist.remove_points(deletemask)
+            
+            # Remove peaks that are too close together
+            if maxNumPeaks is not False:
+                r2 = minPeakSpacing**2
+                deletemask = np.zeros(pointlist.length, dtype=bool)
+                for i in range(pointlist.length):
+                    if deletemask[i] == False:
+                        tooClose = ( (pointlist.data['qx']-pointlist.data['qx'][i])**2 + \
+                                     (pointlist.data['qy']-pointlist.data['qy'][i])**2 ) < r2
+                        tooClose[:i+1] = False
+                        deletemask[tooClose] = True
+                pointlist.remove_points(deletemask)
+
+            # Keep only up to maxNumPeaks
+            if maxNumPeaks is not False:
+                if maxNumPeaks < pointlist.length:
+                    deletemask = np.zeros(pointlist.length, dtype=bool)
+                    deletemask[maxNumPeaks:] = True
+                    pointlist.remove_points(deletemask)
+
+    return pointlistarray
+
 
+def get_pointlistarray_hist(pointlistarray, return_hist = False, bins = 200):
```

Comment:
I'd suggest we add a `plot_hist` flag as well, which defaults to `True`, so that users have the option of (a) JUST getting an array of all the peak intensities, (b) getting that array and plotting it's histogram (default), or (c) getting the array, and getting the histogram values